src/select/__tests__/select.test.tsx

```diff
@@ -450,6 +451,162 @@ describe.each([false, true])('expandToViewport=%s', expandToViewport => {
       wrapper.openDropdown();
       expect(wrapper.findDropdown({ expandToViewport })?.findOpenDropdown()).toBeFalsy();
     });
+
+    describe('Disabled item with reason', () => {
+      beforeEach(() => {
+        jest.spyOn(ReactDOM, 'createPortal').mockImplementation((element: any) => element);
+      });
+      afterEach(() => {
+        jest.restoreAllMocks();
+      });
+
+      test('has no tooltip open by default', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+                disabledReason: 'disabled reason',
+              };
+            }
+
+            return item;
+          }),
+        });
+        wrapper.openDropdown();
+
+        expect(wrapper.findDropdown().findSelectableItem(1)!.findDisabledReason()).toBe(null);
+      });
+
+      test('has no tooltip without disabledReason', () => {
+        const { wrapper } = renderSelect({
+          options: defaultOptions.map((item, index) => {
+            if (index === 0) {
+              return {
+                ...item,
+                disabled: true,
+              };
+            }
+
+            return item;
+          }),
+        });
```

Comment:
I think at this point it is simpler to create your own copy of props than deriving them from defailt.

Compare:

```js
renderSelect({
   option: [{ label: 'Disabled for no reason ', value: 'foo', disabled: true }]
})
``` 

Why is extra complexity?

(same applies to all `defaultOptions.map` in this file)

Reply:
updated